### DISCUSSION TITLE
getStaticPaths and URL's

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#20194] [Help]
Hi all,

I have an app that is currently SSR that I would like to switch to static.

My problem is that I want the displayed URL to show `/product/{slug}` but I also need access to my itemID.

Currently, it seems that I can only pass either the slug or the id, or show a url similar to `/product/{slug}-id` which isn't ideal.

Is there a way to achieve the desired result using Next?

**Comment by stigkj** [comment]
Not quite sure what you are asking about. I guess there is a mapping between the slug and the id? Is that mapping in a db or something? Then you can just do the lookup in `getStaticProps` as that method is run only on the server.

**Reply by darrylmorley** [reply]
I'm trying to achieve pretty URLs. It doesn't appear that this will be achievable though.

**Comment by lalit2005** [comment]
I guess you can use query parameters for the id like `/products/{slug}/q?id=35`

Hope this helps🤞

**Reply by darrylmorley** [reply]
Aye, that's the route I've taken. Thank you.

**Reply by stigkj** [reply]
@darrylmorley To achieve pretty URLs and have a static page, all parameters to the page must be in the URL path, i.e. `id` cannot be a query parameter. So you would need something like this: `/products/{slug}/{id}`.

But if there is a 1-1 connection between `slug` and `id`, can't you lookup one from the other in `getStaticProps`?
